src/LoadingStoreCollection.ts

```diff
@@ -0,0 +1,40 @@
+import LoadingStoreValue from './LoadingStoreValue'
+import Resource from './interfaces/Resource'
+
+class LoadingStoreCollection {
+  /**
+   * Returns a placeholder for an array that has not yet finished loading from the API. The array placeholder
+   * will respond to functional calls (like .find(), .map(), etc.) with further LoadingStoreCollections or
+   * LoadingStoreValues. If passed the existingContent argument, random access and .length will also work.
+   * @param arrayLoaded     Promise that resolves once the array has finished loading
+   * @param existingContent optionally set the elements that are already known, for random access
+   */
+  static create (loadArray: Promise<Array<Resource> | undefined>, existingContent: Array<Resource> = []): Array<Resource> {
```

Comment:
In this case, we either need to wrap it. Or extend the Array class. I tried latter, and somehow failed. Let my try former :dizzy_face:

Reply:
One conclusion from my experiments: Extending the Array class is not possible, because the LoadingStoreCollection has different type signatures for either `find` or `map`:
* If `LoadingStoreCollection` inherits from `Array<Resource>` or some other `Array<Something>`, or we make a generic `LoadingStoreCollection<T>`, the `find` method can no longer return a `LoadingStoreValue`, but must return `Resource` or `Something` or `T`.
* If `LoadingStoreCollection` inherits from `Array<LoadingStoreValue>`, the `map` method cannot be made to work the way we want, because it has to potentially transform the `LoadingStoreValue`s into objects of other classes, depending on the callback passed to `map` (the callback might map every entry to an integer, for example).

Reply:
TODO: Try something like `LoadingStoreValue<T> extends T`, this way the Array contract could maybe hold